nx/lib/nx.ex

```diff
@@ -7390,13 +7429,165 @@ defmodule Nx do
+      iex> Nx.argsort(Nx.tensor([16, 23, 42, 4, 8, 15]))
+      #Nx.Tensor<
+        s64[6]
+        [3, 4, 5, 0, 1, 2]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :x)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 1],
+          [0, 1, 0]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 2],
+          [0, 2, 1]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :y, comparator: :asc)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [2, 0, 1],
+          [1, 2, 0]
+        ]
+      >
+
+      iex> tensor = Nx.tensor([[[4, 5], [2, 5], [5, 0]], [[1, 9], [2, 1], [2, 1]], [[0, -1], [-1, 0], [0, -1]], [[-1, 0], [0, -1], [-1, 0]]], names: [:x, :y, :z])
+      iex> Nx.argsort(tensor, axis: :x)
+      #Nx.Tensor<
+        s64[x: 4][y: 3][z: 2]
+        [
+          [
+            [3, 2],
+            [2, 3],
+            [3, 2]
+          ],
+          [
+            [2, 3],
+            [3, 2],
+            [2, 0]
+          ],
+          [
+            [1, 0],
+            [0, 1],
+            [1, 3]
+          ],
+          [
+            [0, 1],
+            [1, 0],
+            [0, 1]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :x)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [1, 0, 0],
+            [0, 1, 0],
+            [1, 0, 0]
+          ],
+          [
+            [0, 1, 1],
+            [1, 0, 1],
+            [0, 1, 1]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :y)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [1, 2, 0],
+            [0, 0, 2],
+            [2, 1, 1]
+          ],
+          [
+            [0, 1, 1],
+            [1, 2, 2],
+            [2, 0, 0]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[[4, 5, 2], [2, 5, 3], [5, 0, 2]], [[1, 9, 8], [2, 1, 3], [2, 1, 4]]], names: [:x, :y, :z]), axis: :z)
+      #Nx.Tensor<
+        s64[x: 2][y: 3][z: 3]
+        [
+          [
+            [2, 0, 1],
+            [0, 2, 1],
+            [1, 2, 0]
+          ],
+          [
+            [0, 2, 1],
+            [1, 0, 2],
+            [1, 0, 2]
+          ]
+        ]
+      >
+
+      iex> Nx.argsort(Nx.tensor([[3, 1, 7], [2, 5, 4]], names: [:x, :y]), axis: :x, comparator: &Nx.less/2)
+      #Nx.Tensor<
+        s64[x: 2][y: 3]
+        [
+          [1, 0, 1],
+          [0, 1, 0]
+        ]
+      >
+  """
+  @doc type: :ndim
+  def argsort(tensor, opts \\ []) do
+    opts = keyword!(opts, axis: 0, comparator: :desc)
```

Comment:
The default here seems to be `:desc`. We probably want to change both sort and argsort to be asc.

Reply:
It seems that the comparator definitions were swapped between desc and asc. My latest commit changed this